src/mousev1_to_nwb/abdeladim_2023/abdeladim_2023imaginginterface.py

```diff
@@ -0,0 +1,123 @@
+
+    def __init__(
+        self,
+        folder_path: FolderPathType,
+        channel_name: str,
+        plane_name: str,
+        verbose: bool = True,
+    ):
+        self.channel_name = channel_name
+        self.plane_name = plane_name
+
+        file_path = sorted(glob.glob(f"{folder_path}/*.tif"))
+        self.image_metadata = extract_extra_metadata(file_path=file_path[0])
+        super().__init__(folder_path=folder_path,channel_name=channel_name,plane_name=plane_name, verbose=verbose)
+
+    def get_metadata(self) -> DeepDict:
+        device_number = 0  # Imaging plane metadata is a list with metadata for each plane
+        metadata = super().get_metadata()
+        if "state.internal.triggerTimeString" in self.image_metadata:
+            extracted_session_start_time = dateparse(self.image_metadata["state.internal.triggerTimeString"])
+            metadata["NWBFile"].update(session_start_time=extracted_session_start_time)
+        elif "epoch" in self.image_metadata:
+            # Versions of ScanImage at least as recent as 2020, and possibly earlier, store the start time under keyword
+            # `epoch`, as a string encoding of a Matlab array, example `'[2022  8  8 16 56 7.329]'`
+            # dateparse can't cope with this representation, so using strptime directly
+            extracted_session_start_time = datetime.datetime.strptime(
+                self.image_metadata["epoch"], "[%Y %m %d %H %M %S.%f]"
+            )
+            metadata["NWBFile"].update(session_start_time=extracted_session_start_time)
+
+        # Extract many scan image properties and attach them as dic in the description
+        ophys_metadata = metadata["Ophys"]
+        two_photon_series_metadata = ophys_metadata["TwoPhotonSeries"][device_number]
+        if self.image_metadata is not None:
+            extracted_description = json.dumps(self.image_metadata)
+            two_photon_series_metadata.update(description=extracted_description)
+        channel_name_without_space = self.channel_name.replace(" ","")
+        indicators = {"Channel1":"GCaMP6f", "Channel2":"tdTomato"}
+
+        channel_metadata = {
+            "Channel1": {
+                "name": "Green",
+                "emission_lambda": 513.0, #TODO check
+                "description": "Green channel of the microscope, 525/50 nm filter.",#TODO check
+            },
+            "Channel2": {
+                "name": "Red",
+                "emission_lambda": 581.0,#TODO check
+                "description": "Red channel of the microscope, 550/50 nm filter.",#TODO check
+            },
+        }[channel_name_without_space]
+
+        optical_channel_metadata = channel_metadata
+
+        device_name = "ADD_DEVICE_NAME" #TODO add device name
+        metadata["Ophys"]["Device"][0].update(
+            name=device_name, description="ADD_DEVICE_DESCRIPTION", manufacturer="ADD_DEVICE_MANUFACTURER" #TODO add device description and manufacturer
```

Comment:
I think manufacturer is an optional argument in Device, if this microscope is custom built and you don't find this info in the papers then I think it's fine to not specify. just add a description that it is custom built

Reply:
I will change this in #5